tinybmp/src/lib.rs

```diff
@@ -58,4 +58,103 @@ impl<'a> Bmp<'a> {
     pub fn bpp(&self) -> u32 {
         u32::from(self.header.bpp)
     }
+
+    /// Returns the row length in bytes.
+    ///
+    /// Each row in a BMP file is a multiple of 4 bytes long.
+    fn bytes_per_row(&self) -> usize {
+        let bits_per_row = self.width() as usize * self.bpp() as usize;
+
+        (bits_per_row + 31) / 32 * (32 / 8)
+    }
+}
+
+impl<'a> IntoIterator for &'a Bmp<'a> {
+    type Item = u32;
+    type IntoIter = BmpIterator<'a>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        let stride = match self.bpp() {
+            8 => 1,
+            16 => 2,
+            24 => 3,
+            32 => 4,
+            depth => panic!("Bit depth {} not supported", depth),
+        };
+
+        BmpIterator {
+            bmp: self,
+            pixel_data: self.image_data(),
+            stride,
+            x: 0,
+            y: 0,
+            start_idx: 0,
+        }
+    }
+}
+
+/// Iterator over individual BMP pixels
+///
+/// This can be used to build a raw image buffer to pass around
+#[derive(Debug)]
+pub struct BmpIterator<'a> {
+    /// Reference to original BMP image
+    bmp: &'a Bmp<'a>,
+
+    /// Image pixel data as a byte slice, little endian ordering
+    pixel_data: &'a [u8],
+
+    /// Number of bytes per pixel
+    stride: usize,
```

Comment:
I've also seen `stride` used for `bytes_per_row` before, so this could be renamed to `pixel_stride` to make this unambiguous.